File: 1769/solution.go
func minOperations(boxes string) []int {
	result := make([] int, len(boxes))
	
	ops, balls := 0, 0
	for idx, _ := range(boxes) {
			result[idx] += ops
			if boxes[idx] == '1' {
					balls++
			}
			ops += balls
	}
	
	ops, balls = 0, 0
	for i := len(boxes) - 1; i >= 0; i-- {
			result[i] += ops
			if boxes[i] == '1' {
					balls++
			}
			ops += balls
	}
	
	return result
}

File: 1769/solution.cpp
class Solution {
public:
    vector<int> minOperations(string boxes) {
        vector<int> result(boxes.size(), 0);
        
        // -->
        int ops = 0, balls = 0;
        for (int i=0; i<boxes.size(); i++) {
            result[i] += ops;
            balls += (boxes[i] == '1' ? 1 : 0);
            ops += balls;
        }
        
        ops = 0;
        balls = 0;
        for (int i=boxes.size() - 1; i >= 0; i--) {
            result[i] += ops;
            balls += (boxes[i] == '1' ? 1 : 0);
            ops += balls;
        }
        
        return result;
    }
};